service/history/replication/task_executor.go

```diff
@@ -306,19 +306,23 @@ func (e *taskExecutorImpl) filterTask(
 	}
 
 	namespaceEntry, err := e.namespaceRegistry.GetNamespaceByID(namespaceID)
-	if err != nil {
-		return false, err
-	}
-
-	shouldProcessTask := false
-FilterLoop:
-	for _, targetCluster := range namespaceEntry.ClusterNames() {
-		if e.currentCluster == targetCluster {
-			shouldProcessTask = true
-			break FilterLoop
+	switch err.(type) {
```

Comment:
I feel like an existing unit test should fail? Otherwise we probably don't have proper test coverage :)
I stamped this PR but please use your judgement here.

Reply:
Let me add one

Reply:
To me it looks like embedding the `shouldProcessTask` logic in the error handling part makes the code a bit hard to understand. 

(FYI, currently the only error returned by GetNamespaceByID is namespace not found, but that may change in the future if we make to make the cache read through. )

Reply:
Revert to original logic and just handle namespace not found error